src/Games/NexusMods.Games.StardewValley/Emitters/SMAPIGameVersionDiagnosticEmitter.cs

```diff
@@ -0,0 +1,376 @@
+            if (sMaximumGameVersion is not null) if (!TryParseVersion(sMaximumGameVersion, out maximumGameVersion)) continue;
+
+            res[version] = new GameVersions
+            {
+                MinimumGameVersion = minimumGameVersion,
+                MaximumGameVersion = maximumGameVersion,
+            };
+        }
+
+        _smapiToGameMappings = res.ToImmutableDictionary(keyComparer: VersionComparer.Instance);
+        return _smapiToGameMappings;
+
+    }
+
+    private async Task<GameToSMAPIMapping?> FetchGameToSMAPIMappings(CancellationToken cancellationToken)
+    {
+        if (_gameToSMAPIMappings is not null) return _gameToSMAPIMappings;
+
+        var data = await FetchData<Dictionary<string, string[]>>(GameToSMAPIMappingsDataUri, cancellationToken);
+        if (data is null) return null;
+
+        var res = new Dictionary<Version, Version>();
+        foreach (var kv in data)
+        {
+            var (sGameVersion, smapiVersions) = kv;
+            if (smapiVersions.Length == 0) continue;
+
+            if (!TryParseVersion(sGameVersion, out var gameVersion)) continue;
+
+            var sSMAPIVersion = smapiVersions.First();
+            if (!TryParseVersion(sSMAPIVersion, out var smapiVersion)) continue;
+
+            res[gameVersion] = smapiVersion;
+        }
+
+        _gameToSMAPIMappings = res.ToImmutableDictionary(keyComparer: VersionComparer.Instance);
+        return _gameToSMAPIMappings;
+    }
+
+    private bool TryParseVersion(string input, [NotNullWhen(true)] out Version? version)
+    {
+        version = null;
+        if (Version.TryParse(input, out var tmp))
+        {
+            version = SimplifyVersion(tmp);
+            return true;
+        }
+
+        _logger.LogWarning("Unable to parse string `{Input}` as a Version", input);
+        return false;
+    }
+
+    private static Version SimplifyVersion(Version input)
+    {
+        var major = input.Major;
+        var minor = input.Minor;
+        var build = input.Build == -1 ? 0 : input.Build;
+        var revision = input.Revision == 0 ? -1 : input.Revision;
+
+        if (revision != -1) return new Version(major, minor, build, revision);
+        return new Version(major, minor, build);
+    }
+
+    private async Task<T?> FetchData<T>(Uri dataUri, CancellationToken cancellationToken) where T : class
+    {
+        try
+        {
+            var data = await _client.GetFromJsonAsync<T>(dataUri, cancellationToken).ConfigureAwait(false);
+            if (data is not null) return data;
+
+            _logger.LogWarning("Serialization of JSON data at {Uri} failed and returned null", dataUri);
+            return null;
+
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Exception while getting JSON data from {Uri}", dataUri);
+            return null;
+        }
+    }
+
+    private class VersionComparer : IEqualityComparer<Version>, IComparer<Version>
```

Comment:
I would add a comment saying this is skipping the revision number by design as the manifests don't have it